static/js/pages/instances/forms/DevicesForm.tsx

```diff
@@ -0,0 +1,215 @@
+                  onChange={formik.handleChange}
+                  value={(formik.values.devices[index] as LxdNicDevice).network}
+                  options={getNetworkOptions()}
+                />
+                <Input
+                  label="Network name"
+                  name={`devices.${index}.name`}
+                  onBlur={formik.handleBlur}
+                  onChange={formik.handleChange}
+                  value={(formik.values.devices[index] as LxdNicDevice).name}
+                  type="text"
+                  placeholder="Enter name"
+                />
+              </>
+            )}
+            {formik.values.devices[index].type === "disk" && (
+              <>
+                <Select
+                  label="Storage pool"
+                  name={`devices.${index}.pool`}
+                  onBlur={formik.handleBlur}
+                  onChange={formik.handleChange}
+                  value={(formik.values.devices[index] as LxdDiskDevice).pool}
+                  options={getStoragePoolOptions()}
+                />
+                <Input
+                  label="Name"
+                  name={`devices.${index}.name`}
+                  onBlur={formik.handleBlur}
+                  onChange={formik.handleChange}
+                  value={(formik.values.devices[index] as LxdDiskDevice).name}
+                  type="text"
+                  placeholder="Enter a name (i.e. root)"
+                />
+                <Input
+                  label="Path"
+                  name={`devices.${index}.path`}
+                  onBlur={formik.handleBlur}
+                  onChange={formik.handleChange}
+                  value={(formik.values.devices[index] as LxdDiskDevice).path}
+                  type="text"
+                  placeholder="Enter path (i.e. /)"
+                />
+                <Input
+                  label="Size limit in GB"
+                  name={`devices.${index}.size`}
+                  onBlur={formik.handleBlur}
+                  onChange={formik.handleChange}
+                  value={(formik.values.devices[index] as LxdDiskDevice).size}
+                  type="text"
+                  placeholder="Enter number with GB suffix"
+                />
+              </>
+            )}
+            <hr />
+          </Col>
+          <Col size={3}>
+            <Button
+              className="delete-device"
+              onClick={() => removeDiskRow(index)}
+              type="button"
+              appearance="link"
+            >
+              Delete device
+            </Button>
+          </Col>
+        </Row>
+      ))}
+      <Button onClick={addDeviceRow} type="button" hasIcon>
+        <Icon name="plus" />
+        <span>Add device</span>
+      </Button>
+      <Link
+        href="https://linuxcontainers.org/lxd/docs/latest/reference/devices/"
+        target="_blank"
+        rel="noreferrer"
+      >
+        What is a device <Icon name="external-link" />
```

Comment:
The style of the icon is inconsistent with that of the link. What I did for the empty state was to apply the link style to it through a class: https://github.com/canonical/lxd-ui/blob/main/static/sass/_empty_state.scss#L10
However, there is a problem with this: when a link is visited, its text colour changes, but that of the icon does not.

Reply:
I think it is quite common to not color the icon. Especially as the color will differ with a visited link, I lean towards keeping it as it is.

Reply:
@al-bakalova how should we proceed here? I ask this so that we can also adapt the style for empty states, if needed.

Reply:
I'd apply the link style to the icon. After all, the icon itself is also part of the link.